autosklearn/automl.py

```diff
@@ -917,6 +920,160 @@ def refit(self, X, y):
         self._can_predict = True
         return self
 
+    def fit_pipeline(
+        self,
+        X: np.ndarray,
+        y: np.ndarray,
+        task: int,
+        is_classification: bool,
+        **kwargs: Dict,
+    ) -> Tuple[Optional[BasePipeline], RunInfo, RunValue]:
+        """ Fits and individual pipeline configuration and returns
+        the result to the user.
+
+        The Estimator constraints are honored, for example the resampling
+        strategy, or memory constraints, unless directly provided to the method.
+        By default, this method supports the same signature as fit(), and any extra
+        arguments are redirected to the TAE evaluation function, which allows for
+        further customization while building a pipeline.
+
+        Parameters
+        ----------
+            X: array-like, shape = (n_samples, n_features)
+                The features used for training
+            y: array-like
+                The labels used for training
+            task: int
+                The type of task, taken from autosklearn.constants
+            is_classification: bool
+                Whether the task is for classification or regression. This affects
+                how the targets are treated
+
+        Returns
+        -------
+            pipeline: Optional[BasePipeline]
+                The fitted pipeline. In case of failure while fitting the pipeline,
+                a None is returned.
+            run_info: RunInFo
+                A named tuple that contains the configuration launched
+            run_value: RunValue
+                A named tuple that contains the result of the run
+        """
+
+        # Get the configuration space
+        # This also ensures that the Backend has processed the
+        # dataset
+        if self.configuration_space is None:
+            self.configuration_space = self.fit(
+                X=X, y=y, task=task,
+                X_test=kwargs.pop('X_test', None),
+                y_test=kwargs.pop('y_test', None),
+                feat_type=kwargs.pop('feat_type', None),
+                dataset_name=kwargs.pop('dataset_name', None),
+                only_return_configuration_space=True)
+
+        # Get a configuration from the user or sample from the CS
+        config = kwargs.pop('config', self.configuration_space.sample_configuration())
```

Comment:
Could you please describe the use case where a user would like to fit a random configuration?

Reply:
This is no longer necessary if the configuration needs to be provided by the user, right?